test/test_check_db_dump_recent.py

```diff
@@ -0,0 +1,217 @@
+        temp_file, self.temp_file_path = tempfile.mkstemp()
+        os.close(temp_file)
+
+        # Monkey patch the datetime module within check.check_db_dump_recent
+        # so we don't have to worry about hard coded datetimes in these tests.
+        check.check_db_dump_recent.datetime = MonkeyDateTime
+
+    def tearDown(self):
+        os.remove(self.temp_file_path)
+
+    def test_recent_success_mariadb(self):
+        """Tests the check script given a recent, successful, run."""
+        log_line_list = [
+            "2023-01-01T09:40:01+0000",
+            "2023-01-01T09:41:01+0000 ERROR: An Error",
+            "2023-01-01T10:40:01+0000",
+            "2023-01-01T10:41:01+0000 ERROR: An Error",
+            "2023-01-01T10:42:01+0000 ERROR: Another Error",
+            "2023-01-01T11:40:01+0000",
+            "2023-01-01T11:41:01+0000 INFO: completed ok",
+        ]
+
+        self._run_test(log_line_list, check.check_db_dump_recent.RETURN_CODE_OK)
+
+    def test_recent_failure_mariadb(self):
+        """Tests the check script given a recent, unsuccessful, run."""
+        log_line_list = [
+            "2023-01-01T09:40:01+0000",
+            "2023-01-01T09:41:01+0000 ERROR: An Error",
+            "2023-01-01T10:40:01+0000",
+            "2023-01-01T10:41:01+0000 ERROR: An Error",
+            "2023-01-01T10:42:01+0000 ERROR: Another Error",
+            "2023-01-01T11:40:01+0000",
+            "2023-01-01T11:41:01+0000 INFO: completed ok",
+            "2023-01-01T11:50:01+0000",
+            "2023-01-01T11:51:01+0000 ERROR: Yet another Error",
+        ]
+
+        self._run_test(log_line_list, check.check_db_dump_recent.RETURN_CODE_OK)
+
+    def test_old_success_mariadb(self):
+        """Test the check script given an old, successful, run."""
+        log_line_list = [
+            "2023-01-01T09:40:01+0000",
+            "2023-01-01T09:41:01+0000 INFO: completed ok",
+            "2023-01-01T10:40:01+0000",
+            "2023-01-01T10:41:01+0000 ERROR: An Error",
+            "2023-01-01T10:42:01+0000 ERROR: Another Error",
+            "2023-01-01T11:40:01+0000",
+            "2023-01-01T11:41:01+0000 ERROR: Yet Another Error",
+        ]
+
+        self._run_test(log_line_list, check.check_db_dump_recent.RETURN_CODE_CRITICAL)
+
+    def test_never_successful_mariadb(self):
+        """Test the check script given a lack of success, ever."""
+        log_line_list = [
+            "2023-01-01T09:40:01+0000",
+            "2023-01-01T09:41:01+0000 ERROR: An Error",
+            "2023-01-01T10:40:01+0000",
+            "2023-01-01T10:41:01+0000 ERROR: An Error",
+            "2023-01-01T10:42:01+0000 ERROR: Another Error",
+            "2023-01-01T11:40:01+0000",
+            "2023-01-01T11:41:01+0000 ERROR: Yet another Error",
+        ]
+
+        self._run_test(log_line_list, check.check_db_dump_recent.RETURN_CODE_CRITICAL)
+
+    def test_recent_success_oracle(self):
```

Comment:
What's the difference between a mariadb and an oracle test?

Reply:
The input, for Oracle error messages don't have a timestamp associated with them. For MariaDB they do.

Reply:
Greg to add class documentation to tests explaining format of example log files.

Reply:
Done in 529bdf35939aafeb68acc2b36c1d4c9d2d12d857